src/cascadia/TerminalControl/ControlInteractivity.cpp

```diff
@@ -293,15 +293,17 @@ namespace winrt::Microsoft::Terminal::Control::implementation
         {
             if (_singleClickTouchdownPos)
             {
-                // Figure out if the user's moved a quarter of a cell's smaller axis away from the clickdown point
-                auto& touchdownPoint{ *_singleClickTouchdownPos };
-                float dx = ::base::saturated_cast<float>(pixelPosition.x() - touchdownPoint.x());
-                float dy = ::base::saturated_cast<float>(pixelPosition.y() - touchdownPoint.y());
-                auto distance{ std::sqrtf(std::powf(dx, 2) +
-                                          std::powf(dy, 2)) };
-
-                const auto fontSizeInDips{ _core->FontSizeInDips() };
-                if (distance >= (std::min(fontSizeInDips.width(), fontSizeInDips.height()) / 4.f))
+                // Figure out if the user's moved a 1/4th of a cell's smaller axis
+                // (practically always the width) away from the clickdown point.
+                const auto fontSizeInDips = _core->FontSizeInDips();
+                const auto touchdownPoint = *_singleClickTouchdownPos;
+                const auto dx = pixelPosition.x - touchdownPoint.x;
+                const auto dy = pixelPosition.y - touchdownPoint.y;
+                const auto w = fontSizeInDips.width;
+                const auto distanceSquared = dx * dx + dy * dy;
+                const auto maxDistanceSquared = w * w / 16; // (w /4)^2

```

Comment:
so we're using `(w/4)^2` here instead of `(std::min(w, height)/4)^2` because w is always less than height?

Reply:
Yeah it just simplifies the code. I hoped that the comment above this code block explains it sufficiently.
Apart from that this is a bog-standard distance check for CPUs. It's not about speed necessarily, it's just that the `sqrt` function is weird, whacky, and comes with a whole baggage of stuff we don't need if we just square the distance instead.